File: ECommerceAPI/ECommerceAPI/Base/Localization/Models/Culture.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ECommerceAPI.Base.Localization.Models
{
    public class Culture
    {
        public int CultureId { get; set; }
        public string Name { get; set; }
        public string Flag { get; set; }
        public string Value { get; set; }
        public bool Active { get; set; }
        public Dictionary<string, string> LocalizationPair { get; set; }

        public Culture()
        {
            LocalizationPair = new Dictionary<string, string>();
        }

        public Culture(string name, string value, string flag)
        {
            this.Name = name;
            this.Value = value;
            this.Flag = flag;
        }

        public Culture Clear()
        {
            return new Culture();
        }
    }
}

File: ECommerceFront/src/app/common/models/culture.ts
import { LocalizationEntity } from './localizationEntity';

export class Culture {
    cultureId: number;
    name: string;
    value: string;
    flag: string;
    localizationPair: any;
    active: boolean;

    constructor(name: string, value: string, flag: string) {
        this.name = name;
        this.value = value;
        this.flag = flag;
        this.localizationPair = [];
    }
}
